andela/bp-esa-backend
# slack service to remove user from slack channels

Since the addition or removal of user(s) from a slack channel is supposed to capture a process, e.g (staffing or rolloff), this pull request demonstrates the process rather than a single instance of adding or removing user to or from a slack channel.

**NOTE:** I've used my own personal slack workspace since the original workspace, that is `ESA-test-workspace` that was originally used for testing this functionality had restrictions from the workspace creator.

**How to Test**
1. Join my [personal workspace](https://alchemist09.slack.com) on slack 
2. Create multiple private channel to showcase a typical staffing scenario. For example, you can create two channels like `a-devs` for available developers and `p-ptr` for a partner channel. 
3. Open the file `core_channels.js` within the routes folder and substitute the values of the core_channels object with your own pseudo values that match the names of the private channels you created. Make sure you don't change the keys. Don't add the p-partner channel you created within the object since it will be passed with the request.

4. Using Postman, make a POST request to the endpoint **/staffing/staffed, with the request body containing matching:

```
{
    channel: `p-ptr`,
        emails: [
            fname.lname@andela.com,
            user2@gmail.com,
        ]
}
```

Make sure you substitute the channel `p-ptr` and values in the `emails` array above with your own valid values for the request to be valid.

5. Check if user(s) have been removed from and added to the respective channels that match the intended purposes when created. For the example above, users with the emails passed in the request body should be removed from `a-devs`, the placeholder available-developers channel, and added to `p-ptr`, the placeholder partner channel.

6. Using Postman, make a POST request the endpoint **/staffing/rolloff** with a request body as step four above.
7. Check respective channels to confirm if users have been added to or removed from the intended channels. In the example above, the users matching the emails passed in the `emails` array should be removed from `p-ptr` channel and added back to `a-devs` channel.

Fixes issue #2 #19
